metricbeat/helper/privileges_windows.go

```diff
@@ -0,0 +1,83 @@
+package helper
+
+import (
+	"syscall"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/gosigar/sys/windows"
+	"github.com/pkg/errors"
+)
+
+// errMissingSeDebugPrivilege indicates that the SeDebugPrivilege is not
+// present in the process's token. This is distinct from disabled. The token
+// would be missing if the user does not have "Debug programs" rights. By
+// default, only administrators and LocalSystem accounts have the privileges to
+// debug programs.
+var errMissingSeDebugPrivilege = errors.New("Metricbeat is running without " +
+	"SeDebugPrivilege, a Windows privilege that allows it to collect metrics " +
+	"from other processes. The user running Metricbeat may not have the " +
+	"appropriate privileges or the security policy disallows it.")
+
+// enableSeDebugPrivilege enables the SeDebugPrivilege if it is present in
+// the process's token.
+func enableSeDebugPrivilege() error {
+	self, err := syscall.GetCurrentProcess()
+	if err != nil {
+		return err
+	}
+
+	var token syscall.Token
+	err = syscall.OpenProcessToken(self, syscall.TOKEN_QUERY|syscall.TOKEN_ADJUST_PRIVILEGES, &token)
+	if err != nil {
+		return err
+	}
+
+	if err = windows.EnableTokenPrivileges(token, windows.SeDebugPrivilege); err != nil {
+		return errors.Wrap(err, "EnableTokenPrivileges failed")
+	}
+
+	return nil
+}
+
+// CheckAndEnableSeDebugPrivilege checks if the process's token has the
+// SeDebugPrivilege and enables it if it is disabled.
+func CheckAndEnableSeDebugPrivilege() error {
```

Comment:
I'm thinking this should be `checkAndEnableSeDebugPrivilege()` and a separate method named `CheckAndEnableSeDebugPrivilege` exists that calls `checkAndEnableSeDebugPrivilege()` only once by using `sync.Once`.

This way only the system module or the windows module will do the initialization, but not both.